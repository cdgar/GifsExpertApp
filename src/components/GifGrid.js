import React from 'react';
import useFetchGifs from '../hooks/useFetchGifs'
import GifGridItem from './GifGridItem';
import PropTypes from 'prop-types';

const GifGrid = ({category}) => {

    const {data: images, loading} = useFetchGifs(category);

    return (
        <>
            <h3>{category}</h3>

            {loading && <p className="animate__animated animate__bounceOutRight">Loading</p>}
            <div className="card-grid">
                {
                    images.map((i) => (
                        <GifGridItem 
                            key={i.id} 
                            {...i}
                        />
                    ))
                }
            </div>
        </>
    )
}

GifGrid.propTypes = {
    category: PropTypes.string.isRequired
}

export default GifGrid;

