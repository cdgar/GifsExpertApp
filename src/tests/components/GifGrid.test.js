import React from 'react';
import { shallow } from "enzyme"
import GifGrid from '../../components/GifGrid';
import useFetchGifs from '../../hooks/useFetchGifs';

jest.mock('../../hooks/useFetchGifs');

describe('GifGrid.js', () => {
    const category = 'dragon';
    useFetchGifs.mockReturnValue({
        data: [], 
        loading: true
    });
    let wrapper = shallow(<GifGrid category={category}/>);
    
    beforeEach(() => {
        useFetchGifs.mockReturnValue({
            data: [], 
            loading: true
        });
        wrapper = shallow(<GifGrid category={category}/>);
    });

    test('should show GifGrid', () => {
        expect(wrapper).toMatchSnapshot();
    })

    test('should show some item when load images', () => {
        const gifs= [{
            id: 'id',
            title: 'img.title',
            url: 'img.images.downsized_medium.url'
        },{
            id: 'id',
            title: 'img.title',
            url: 'img.images.downsized_medium.url'
        }];

        useFetchGifs.mockReturnValue({
            data: gifs, 
            loading: false
        });
        wrapper = shallow(<GifGrid category={category}/>);

        // expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('p').exists()).toBe(false);
        expect(wrapper.find('GifGridItem').length).toBe(gifs.length);
    })

})