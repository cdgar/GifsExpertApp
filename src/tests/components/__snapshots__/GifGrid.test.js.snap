// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GifGrid.js should show GifGrid 1`] = `
<Fragment>
  <h3>
    dragon
  </h3>
  <p
    className="animate__animated animate__bounceOutRight"
  >
    Loading
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;
